feature/widget/src/main/java/com/twofasapp/feature/widget/ui/settings/WidgetSettingsViewModel.kt

```diff
@@ -59,12 +68,13 @@ class WidgetSettingsViewModel(
             widgetsRepository.saveWidget(
                 widget.copy(
                     lastInteraction = timeProvider.systemCurrentTime(),
-                    services = uiState.value.selected.map { serviceId ->
-                        WidgetService(
-                            service = uiState.value.services.first { it.id == serviceId },
-                            revealed = false,
-                        )
-                    }
+                    services = uiState.value.services.filter { uiState.value.selected.contains(it.id) }
+                        .map { service ->
+                            WidgetService(
+                                service = service,
+                                revealed = false,
+                            )
+                        },
```

Comment:
We don't need to keep the `selected` list's sorting in-sync with `services` until actually saving the widget. Filtering here makes the list sent to the widget inherit the sorting of the whole list.